pkg/utils/utils.go

```diff
@@ -117,3 +122,40 @@ func NewGenericTestAndValidate(templateFile, schemaPath string, values map[strin
 
 	return tester, handlers
 }
+
+// RunCommandInContainerNameSpace run a host command in a running container with the nsenter command.
+// takes the container nodeName, node Oc and container UID
+// returns the raw output of the command
+func RunCommandInContainerNameSpace(nodeName string, nodeOc *interactive.Oc, containerID, command string, timeout time.Duration, isNonOcp bool) string {
+	containrPID := GetContainerPID(nodeName, nodeOc, containerID, isNonOcp)
+	nodeCommand := "nsenter -t " + containrPID + " -n " + command
+	return RunCommandInNode(nodeName, nodeOc, nodeCommand, timeout)
+}
+
+// GetContainerPID gets the container PID from a kubernetes node, Oc and container PID
+func GetContainerPID(nodeName string, nodeOc *interactive.Oc, containerID string, isNonOcp bool) string {
+	command := ""
+	if isNonOcp {
+		command = "chroot /host docker inspect -f '{{.State.Pid}}' " + containerID + " 2>/dev/null"
```

Comment:
this won't work for containerd. the equivalent command is `ctr --namespace k8s.io containers info 614607bb9bdad5030580fbdfc73d6131441a884a51380c6da51897e8e2c4a3b5` and i don't see pid for some reason

Reply:
```root@cnf-cert-testbed-worker2:/# ctr --namespace k8s.io containers info 614607bb9bdad5030580fbdfc73d6131441a884a51380c6da51897e8e2c4a3b5 | more                                          
{
    "ID": "614607bb9bdad5030580fbdfc73d6131441a884a51380c6da51897e8e2c4a3b5",
    "Labels": {
        "io.cri-containerd.kind": "container",
        "io.kubernetes.container.name": "test",
        "io.kubernetes.pod.name": "test-6bcf66694-m76hq",
        "io.kubernetes.pod.namespace": "tnf",
        "io.kubernetes.pod.uid": "dd089926-9930-4727-a832-93f100743609"
    },
    "Image": "quay.io/testnetworkfunction/cnf-test-partner:latest",
    "Runtime": {
        "Name": "io.containerd.runc.v2",
        "Options": null
    },
    "SnapshotKey": "614607bb9bdad5030580fbdfc73d6131441a884a51380c6da51897e8e2c4a3b5",
    "Snapshotter": "overlayfs",
    "CreatedAt": "2021-12-07T21:01:59.084661017Z",
    "UpdatedAt": "2021-12-07T21:01:59.084661017Z",
    "Extensions": {
        "io.cri-containerd.container.metadata": {
            "type_url": "github.com/containerd/cri/pkg/store/container/Metadata",
            "value": "eyJWZXJzaW9uIjoidjEiLCJNZXRhZGF0YSI6eyJJRCI6IjYxNDYwN2JiOWJkYWQ1MDMwNTgwZmJkZmM3M2Q2MTMxNDQxYTg4NGE1MTM4MGM2ZGE1MTg5N2U4ZTJjNGEzYjUiLCJOYW1lIjoidGVzdF90ZXN0LTZiY2Y2NjY5NC1tNzZocV90bmZfZGQwODk5MjYtOTkzMC00NzI3LWE4MzItOTN
mMTAwNzQzNjA5XzAiLCJTYW5kYm94SUQiOiIwYmUzYTRhZWI4ZTI4MTMyOTAyOTY5YjUzYWI4NWQ3NzgyOTdlMGRjY2UzYzFjNTE2ZDM4NmFmNmRkMDk4MTBjIiwiQ29uZmlnIjp7Im1ldGFkYXRhIjp7Im5hbWUiOiJ0ZXN0In0sImltYWdlIjp7ImltYWdlIjoic2hhMjU2OjNiM2QwZWQ4MzBjZjkzOTA4MGYzYzdhZTUx
ODRhMGViNWI0ZjdkNzgyY2Q4MDcyZTIyNTBiZjNlODM3M2U5ZDQifSwiY29tbWFuZCI6WyIvYmluL2Jhc2giLCItYyIsImVjaG8gJ2xvZ3MnIFx1MDAyNlx1MDAyNiB0YWlsIC1mIC9kZXYvbnVsbCJdLCJlbnZzIjpbeyJrZXkiOiJOR0lOWF9PUEVSQVRPUl9DT05UUk9MTEVSX01BTkFHRVJfTUVUUklDU19TRVJWSUNFX
1NFUlZJQ0VfSE9TVCIsInZhbHVlIjoiMTAuOTYuODcuMjU1In0seyJrZXkiOiJOR0lOWF9PUEVSQVRPUl9DT05UUk9MTEVSX01BTkFHRVJfTUVUUklDU19TRVJWSUNFX1BPUlRfODQ0M19UQ1AiLCJ2YWx1ZSI6InRjcDovLzEwLjk2Ljg3LjI1NTo4NDQzIn0seyJrZXkiOiJOR0lOWF9PUEVSQVRPUl9DT05UUk9MTEVSX0
1BTkFHRVJfTUVUUklDU19TRVJWSUNFX1BPUlRfODQ0M19UQ1BfQUREUiIsInZhbHVlIjoiMTAuOTYuODcuMjU1In0seyJrZXkiOiJLVUJFUk5FVEVTX1NFUlZJQ0VfUE9SVCIsInZhbHVlIjoiNDQzIn0seyJrZXkiOiJOR0lOWF9PUEVSQVRPUl9DT05UUk9MTEVSX01BTkFHRVJfTUVUUklDU19TRVJWSUNFX1BPUlQiLCJ
2YWx1ZSI6InRjcDovLzEwLjk2Ljg3LjI1NTo4NDQzIn0seyJrZXkiOiJOR0lOWF9PUEVSQVRPUl9DT05UUk9MTEVSX01BTkFHRVJfTUVUUklDU19TRVJWSUNFX1BPUlRfODQ0M19UQ1BfUFJPVE8iLCJ2YWx1ZSI6InRjcCJ9LHsia2V5IjoiS1VCRVJORVRFU19TRVJWSUNFX1BPUlRfSFRUUFMiLCJ2YWx1ZSI6IjQ0MyJ9
LHsia2V5IjoiS1VCRVJORVRFU19QT1JUXzQ0M19UQ1BfUFJPVE8iLCJ2YWx1ZSI6InRjcCJ9LHsia2V5IjoiS1VCRVJORVRFU19QT1JUXzQ0M19UQ1BfQUREUiIsInZhbHVlIjoiMTAuOTYuMC4xIn0seyJrZXkiOiJOR0lOWF9PUEVSQVRPUl9DT05UUk9MTEVSX01BTkFHRVJfTUVUUklDU19TRVJWSUNFX1NFUlZJQ0VfU
E9SVF9IVFRQUyIsInZhbHVlIjoiODQ0MyJ9LHsia2V5IjoiS1VCRVJORVRFU19TRVJWSUNFX0hPU1QiLCJ2YWx1ZSI6IjEwLjk2LjAuMSJ9LHsia2V5IjoiS1VCRVJORVRFU19QT1JUIiwidmFsdWUiOiJ0Y3A6Ly8xMC45Ni4wLjE6NDQzIn0seyJrZXkiOiJLVUJFUk5FVEVTX1BPUlRfNDQzX1RDUF9QT1JUIiwidmFsdW
UiOiI0NDMifSx7ImtleSI6Ik5HSU5YX09QRVJBVE9SX0NPTlRST0xMRVJfTUFOQUdFUl9NRVRSSUNTX1NFUlZJQ0VfU0VSVklDRV9QT1JUIiwidmFsdWUiOiI4NDQzIn0seyJrZXkiOiJOR0lOWF9PUEVSQVRPUl9DT05UUk9MTEVSX01BTkFHRVJfTUVUUklDU19TRVJWSUNFX1BPUlRfODQ0M19UQ1BfUE9SVCIsInZhbHV
lIjoiODQ0MyJ9LHsia2V5IjoiS1VCRVJORVRFU19QT1JUXzQ0M19UQ1AiLCJ2YWx1ZSI6InRjcDovLzEwLjk2LjAuMTo0NDMifV0sIm1vdW50cyI6W3siY29udGFpbmVyX3BhdGgiOiIvdmFyL3J1bi9zZWNyZXRzL2t1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQiLCJob3N0X3BhdGgiOiIvdmFyL2xpYi9rdWJlbGV0
L3BvZHMvZGQwODk5MjYtOTkzMC00NzI3LWE4MzItOTNmMTAwNzQzNjA5L3ZvbHVtZXMva3ViZXJuZXRlcy5pb35wcm9qZWN0ZWQva3ViZS1hcGktYWNjZXNzLTJxeGw1IiwicmVhZG9ubHkiOnRydWV9LHsiY29udGFpbmVyX3BhdGgiOiIvZXRjL2hvc3RzIiwiaG9zdF9wYXRoIjoiL3Zhci9saWIva3ViZWxldC9wb2RzL
2RkMDg5OTI2LTk5MzAtNDcyNy1hODMyLTkzZjEwMDc0MzYwOS9ldGMtaG9zdHMifSx7ImNvbnRhaW5lcl9wYXRoIjoiL2Rldi90ZXJtaW5hdGlvbi1sb2ciLCJob3N0X3BhdGgiOiIvdmFyL2xpYi9rdWJlbGV0L3BvZHMvZGQwODk5MjYtOTkzMC00NzI3LWE4MzItOTNmMTAwNzQzNjA5L2NvbnRhaW5lcnMvdGVzdC9jOW
QyZmQ0NyJ9XSwibGFiZWxzIjp7ImlvLmt1YmVybmV0ZXMuY29udGFpbmVyLm5hbWUiOiJ0ZXN0IiwiaW8ua3ViZXJuZXRlcy5wb2QubmFtZSI6InRlc3QtNmJjZjY2Njk0LW03NmhxIiwiaW8ua3ViZXJuZXRlcy5wb2QubmFtZXNwYWNlIjoidG5mIiwiaW8ua3ViZXJuZXRlcy5wb2QudWlkIjoiZGQwODk5MjYtOTkzMC0
0NzI3LWE4MzItOTNmMTAwNzQzNjA5In0sImFubm90YXRpb25zIjp7ImlvLmt1YmVybmV0ZXMuY29udGFpbmVyLmhhc2giOiI4ZmQ1ZGFjZCIsImlvLmt1YmVybmV0ZXMuY29udGFpbmVyLnByZVN0b3BIYW5kbGVyIjoie1wiZXhlY1wiOntcImNvbW1hbmRcIjpbXCIvYmluL3NoXCIsXCItY1wiLFwia2lsbGFsbCAtMCB0
YWlsXCJdfX0iLCJpby5rdWJlcm5ldGVzLmNvbnRhaW5lci5yZXN0YXJ0Q291bnQiOiIwIiwiaW8ua3ViZXJuZXRlcy5jb250YWluZXIudGVybWluYXRpb25NZXNzYWdlUGF0aCI6Ii9kZXYvdGVybWluYXRpb24tbG9nIiwiaW8ua3ViZXJuZXRlcy5jb250YWluZXIudGVybWluYXRpb25NZXNzYWdlUG9saWN5IjoiRmlsZ
SIsImlvLmt1YmVybmV0ZXMucG9kLnRlcm1pbmF0aW9uR3JhY2VQZXJpb2QiOiIzMCJ9LCJsb2dfcGF0aCI6InRlc3QvMC5sb2ciLCJsaW51eCI6eyJyZXNvdXJjZXMiOnsiY3B1X3BlcmlvZCI6MTAwMDAwLCJjcHVfcXVvdGEiOjI1MDAwLCJjcHVfc2hhcmVzIjoyNTYsIm1lbW9yeV9saW1pdF9pbl9ieXRlcyI6NTM2OD
cwOTEyLCJvb21fc2NvcmVfYWRqIjotOTk3LCJodWdlcGFnZV9saW1pdHMiOlt7InBhZ2Vfc2l6ZSI6IjJNQiJ9LHsicGFnZV9zaXplIjoiMUdCIn1dfSwic2VjdXJpdHlfY29udGV4dCI6eyJuYW1lc3BhY2Vfb3B0aW9ucyI6eyJwaWQiOjF9LCJydW5fYXNfdXNlciI6e30sIm1hc2tlZF9wYXRocyI6WyIvcHJvYy9hY3B
pIiwiL3Byb2Mva2NvcmUiLCIvcHJvYy9rZXlzIiwiL3Byb2MvbGF0ZW5jeV9zdGF0cyIsIi9wcm9jL3RpbWVyX2xpc3QiLCIvcHJvYy90aW1lcl9zdGF0cyIsIi9wcm9jL3NjaGVkX2RlYnVnIiwiL3Byb2Mvc2NzaSIsIi9zeXMvZmlybXdhcmUiXSwicmVhZG9ubHlfcGF0aHMiOlsiL3Byb2MvYXNvdW5kIiwiL3Byb2Mv
YnVzIiwiL3Byb2MvZnMiLCIvcHJvYy9pcnEiLCIvcHJvYy9zeXMiLCIvcHJvYy9zeXNycS10cmlnZ2VyIl0sInNlY2NvbXAiOnsicHJvZmlsZV90eXBlIjoxfX19fSwiSW1hZ2VSZWYiOiJzaGEyNTY6M2IzZDBlZDgzMGNmOTM5MDgwZjNjN2FlNTE4NGEwZWI1YjRmN2Q3ODJjZDgwNzJlMjI1MGJmM2U4MzczZTlkNCIsI
kxvZ1BhdGgiOiIvdmFyL2xvZy9wb2RzL3RuZl90ZXN0LTZiY2Y2NjY5NC1tNzZocV9kZDA4OTkyNi05OTMwLTQ3MjctYTgzMi05M2YxMDA3NDM2MDkvdGVzdC8wLmxvZyIsIlN0b3BTaWduYWwiOiIiLCJQcm9jZXNzTGFiZWwiOiIifX0="
        }
    },
    "Spec": {
        "ociVersion": "1.0.2-dev",
        "process": {
            "user": {
                "uid": 0,
                "gid": 0
            },
            "args": [
                "/bin/bash",
                "-c",
                "echo 'logs' \u0026\u0026 tail -f /dev/null"
            ],
            "env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "HOSTNAME=test-6bcf66694-m76hq",
                "container=oci",
                "NGINX_OPERATOR_CONTROLLER_MANAGER_METRICS_SERVICE_SERVICE_HOST=10.96.87.255",
                "NGINX_OPERATOR_CONTROLLER_MANAGER_METRICS_SERVICE_PORT_8443_TCP=tcp://10.96.87.255:8443",
                "NGINX_OPERATOR_CONTROLLER_MANAGER_METRICS_SERVICE_PORT_8443_TCP_ADDR=10.96.87.255",
                "KUBERNETES_SERVICE_PORT=443",
                "NGINX_OPERATOR_CONTROLLER_MANAGER_METRICS_SERVICE_PORT=tcp://10.96.87.255:8443",
                "NGINX_OPERATOR_CONTROLLER_MANAGER_METRICS_SERVICE_PORT_8443_TCP_PROTO=tcp",
                "KUBERNETES_SERVICE_PORT_HTTPS=443",
                "KUBERNETES_PORT_443_TCP_PROTO=tcp",
                "KUBERNETES_PORT_443_TCP_ADDR=10.96.0.1",
                "NGINX_OPERATOR_CONTROLLER_MANAGER_METRICS_SERVICE_SERVICE_PORT_HTTPS=8443",
                "KUBERNETES_SERVICE_HOST=10.96.0.1",
                "KUBERNETES_PORT=tcp://10.96.0.1:443",
                "KUBERNETES_PORT_443_TCP_PORT=443",
                "NGINX_OPERATOR_CONTROLLER_MANAGER_METRICS_SERVICE_SERVICE_PORT=8443",
                "NGINX_OPERATOR_CONTROLLER_MANAGER_METRICS_SERVICE_PORT_8443_TCP_PORT=8443",
                "KUBERNETES_PORT_443_TCP=tcp://10.96.0.1:443"
            ],
            "cwd": "/",
            "capabilities": {
                "bounding": [
                    "CAP_CHOWN",
                    "CAP_DAC_OVERRIDE",
                    "CAP_FSETID",
                    "CAP_FOWNER",
                    "CAP_MKNOD",
                    "CAP_NET_RAW",
                    "CAP_SETGID",
                    "CAP_SETUID",
                    "CAP_SETFCAP",
                    "CAP_SETPCAP",
                    "CAP_NET_BIND_SERVICE",
                    "CAP_SYS_CHROOT",
                    "CAP_KILL",
                    "CAP_AUDIT_WRITE"
                ],
                "effective": [
                    "CAP_CHOWN",
                    "CAP_DAC_OVERRIDE",
                    "CAP_FSETID",
                    "CAP_FOWNER",
                    "CAP_MKNOD",
                    "CAP_NET_RAW",
                    "CAP_SETGID",
                    "CAP_SETUID",
                    "CAP_SETFCAP",
                    "CAP_SETPCAP",
                    "CAP_NET_BIND_SERVICE",
                    "CAP_SYS_CHROOT",
                    "CAP_KILL",
                    "CAP_AUDIT_WRITE"
                ],
                "inheritable": [
                    "CAP_CHOWN",
                    "CAP_DAC_OVERRIDE",
                    "CAP_FSETID",
                    "CAP_FOWNER",
                    "CAP_MKNOD",
                    "CAP_NET_RAW",
                    "CAP_SETGID",
                    "CAP_SETUID",
                    "CAP_SETFCAP",
                    "CAP_SETPCAP",
                    "CAP_NET_BIND_SERVICE",
                    "CAP_SYS_CHROOT",
                    "CAP_KILL",
                    "CAP_AUDIT_WRITE"
                ],
                "permitted": [
                    "CAP_CHOWN",
                    "CAP_DAC_OVERRIDE",
                    "CAP_FSETID",
                    "CAP_FOWNER",
                    "CAP_MKNOD",
                    "CAP_NET_RAW",
                    "CAP_SETGID",
                    "CAP_SETUID",
                    "CAP_SETFCAP",
                    "CAP_SETPCAP",
                    "CAP_NET_BIND_SERVICE",
                    "CAP_SYS_CHROOT",
                    "CAP_KILL",
                    "CAP_AUDIT_WRITE"
                ]
            },
            "oomScoreAdj": -997
        },
        "root": {
            "path": "rootfs"
        },
        "mounts": [
            {
                "destination": "/proc",
                "type": "proc",
                "source": "proc",
                "options": [
                    "nosuid",
                    "noexec",
                    "nodev"
                ]
            },
            {
                "destination": "/dev",
                "type": "tmpfs",
                "source": "tmpfs",
                "options": [
                    "nosuid",
                    "strictatime",
                    "mode=755",
                    "size=65536k"
                ]
            },
            {
                "destination": "/dev/pts",
                "type": "devpts",
                "source": "devpts",
                "options": [
                    "nosuid",
                    "noexec",
                    "newinstance",
                    "ptmxmode=0666",
                    "mode=0620",
                    "gid=5"
                ]
            },
            {
                "destination": "/dev/mqueue",
                "type": "mqueue",
                "source": "mqueue",
                "options": [
                    "nosuid",
                    "noexec",
                    "nodev"
                ]
            },
            {
                "destination": "/sys",
                "type": "sysfs",
                "source": "sysfs",
                "options": [
                    "nosuid",
                    "noexec",
                    "nodev",
                    "ro"
                ]
            },
            {
                "destination": "/sys/fs/cgroup",
                "type": "cgroup",
                "source": "cgroup",
                "options": [
                    "nosuid",
                    "noexec",
                    "nodev",
                    "relatime",
                    "ro"
                ]
            },
            {
                "destination": "/etc/hosts",
                "type": "bind",
                "source": "/var/lib/kubelet/pods/dd089926-9930-4727-a832-93f100743609/etc-hosts",
                "options": [
                    "rbind",
                    "rprivate",
                    "rw"
                ]
            },
            {
                "destination": "/dev/termination-log",
                "type": "bind",
                "source": "/var/lib/kubelet/pods/dd089926-9930-4727-a832-93f100743609/containers/test/c9d2fd47",
                "options": [
                    "rbind",
                    "rprivate",
                    "rw"
                ]
            },
            {
                "destination": "/etc/hostname",
                "type": "bind",
                "source": "/var/lib/containerd/io.containerd.grpc.v1.cri/sandboxes/0be3a4aeb8e28132902969b53ab85d778297e0dcce3c1c516d386af6dd09810c/hostname",
                "options": [
                    "rbind",
                    "rprivate",
                    "rw"
                ]
            },
            {
                "destination": "/etc/resolv.conf",
                "type": "bind",
                "source": "/var/lib/containerd/io.containerd.grpc.v1.cri/sandboxes/0be3a4aeb8e28132902969b53ab85d778297e0dcce3c1c516d386af6dd09810c/resolv.conf",
                "options": [
                    "rbind",
                    "rprivate",
                    "rw"
                ]
            },
            {
                "destination": "/dev/shm",
                "type": "bind",
                "source": "/run/containerd/io.containerd.grpc.v1.cri/sandboxes/0be3a4aeb8e28132902969b53ab85d778297e0dcce3c1c516d386af6dd09810c/shm",
                "options": [
                    "rbind",
                    "rprivate",
                    "rw"
                ]
            },
            {
                "destination": "/var/run/secrets/kubernetes.io/serviceaccount",
                "type": "bind",
                "source": "/var/lib/kubelet/pods/dd089926-9930-4727-a832-93f100743609/volumes/kubernetes.io~projected/kube-api-access-2qxl5",
                "options": [
                    "rbind",
                    "rprivate",
                    "ro"
                ]
            }
        ],
        "annotations": {
            "io.kubernetes.cri.container-name": "test",
            "io.kubernetes.cri.container-type": "container",
            "io.kubernetes.cri.image-name": "quay.io/testnetworkfunction/cnf-test-partner:latest",
            "io.kubernetes.cri.sandbox-id": "0be3a4aeb8e28132902969b53ab85d778297e0dcce3c1c516d386af6dd09810c",
            "io.kubernetes.cri.sandbox-name": "test-6bcf66694-m76hq",
            "io.kubernetes.cri.sandbox-namespace": "tnf"
        },
        "linux": {
            "resources": {
                "devices": [
                    {
                        "allow": false,
                        "access": "rwm"
                    }
                ],
                "memory": {
                    "limit": 536870912
                },
                "cpu": {
                    "shares": 256,
                    "quota": 25000,
                    "period": 100000
                }
            },
            "cgroupsPath": "/kubelet/kubepods/poddd089926-9930-4727-a832-93f100743609/614607bb9bdad5030580fbdfc73d6131441a884a51380c6da51897e8e2c4a3b5",
            "namespaces": [
                {
                    "type": "pid"
                },
                {
                    "type": "ipc",
                    "path": "/proc/306657/ns/ipc"
                },
                {
                    "type": "uts",
                    "path": "/proc/306657/ns/uts"
                },
                {
                    "type": "mount"
                },
                {
                    "type": "network",
                    "path": "/proc/306657/ns/net"
                },
                {
                    "type": "cgroup"
                }
            ],
            "maskedPaths": [
                "/proc/acpi",
                "/proc/kcore",
                "/proc/keys",
                "/proc/latency_stats",
                "/proc/timer_list",
                "/proc/timer_stats",
                "/proc/sched_debug",
                "/proc/scsi",
                "/sys/firmware"
            ],
            "readonlyPaths": [
                "/proc/asound",
                "/proc/bus",
                "/proc/fs",
                "/proc/irq",
                "/proc/sys",
                "/proc/sysrq-trigger"
            ]
        }
    }
}```